tests/support/Python/Test_Machines.py

```diff
@@ -0,0 +1,28 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import os
+import unittest
+import yaml
+from spectre.support.Machines import this_machine, UnknownMachineError
+from spectre.Informer import unit_test_build_path
+
+
+class TestMachines(unittest.TestCase):
+    def setUp(self):
+        self.machinefile_path = os.path.join(unit_test_build_path(),
+                                             'TestMachine.yaml')
+        yaml.safe_dump(
+            dict(Machine=dict(Name="TestMachine",
+                              Description="Just for testing")),
```

Comment:
Maybe also add a test where you set one of the fields that would have been `None`?